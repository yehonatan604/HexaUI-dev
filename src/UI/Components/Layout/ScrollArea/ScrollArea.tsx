import useTheme from "../../../../Core/Context/ThemeContext/hooks/useTheme";
import { TScrollArea } from "./types/TScrollArea";

const ScrollArea = (props: TScrollArea) => {
  const { children, className, maxHeight, ...componentProps } = props;
  const { mode } = useTheme();
  // Options
  const customMaxHeight = maxHeight ? `h-${maxHeight}` : "h-screen";
  const customScrollbarClass = mode === "dark" ? `scrollbar-dark` : `scrollbar-light`;

  // JSX
  return (
    <div
      className={`${customMaxHeight} overflow-auto ${customScrollbarClass} ${className}`}
      {...componentProps}
    >
      {children}
    </div>
  );
};

export default ScrollArea;
