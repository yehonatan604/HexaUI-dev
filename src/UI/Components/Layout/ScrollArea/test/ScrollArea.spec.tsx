import { render, screen } from "@testing-library/react";
import ScrollArea from "../ScrollArea"; // Adjust the import path
import { describe, it, expect, vi, beforeEach } from "vitest";

// Mock the useTheme hook
let mockTheme = { mode: "light" }; // Default mock to light mode

vi.mock("../../../../Core/Context/ThemeContext/hooks/useTheme", () => ({
  __esModule: true,
  default: () => mockTheme, // Return the current mockTheme value
}));

describe("ScrollArea Component", () => {
  beforeEach(() => {
    mockTheme = { mode: "light" }; // Reset to light mode before each test
  });

  it("should render children correctly", () => {
    render(
      <ScrollArea maxHeight="96">
        <div>Child 1</div>
        <div>Child 2</div>
      </ScrollArea>
    );

    // Check if both children are rendered
    expect(screen.getByText("Child 1")).toBeInTheDocument();
    expect(screen.getByText("Child 2")).toBeInTheDocument();
  });

  it("should apply the correct max height class", () => {
    render(
      <ScrollArea maxHeight="96">
        <div>Child</div>
      </ScrollArea>
    );

    const scrollAreaElement = screen.getByText("Child").parentElement;

    // Check if the correct height class is applied
    expect(scrollAreaElement).toHaveClass("h-96");
  });

  it("should apply the correct scrollbar class based on theme mode (light)", () => {
    render(
      <ScrollArea maxHeight="96">
        <div>Light Theme Child</div>
      </ScrollArea>
    );

    const scrollAreaElement = screen.getByText("Light Theme Child").parentElement;

    // Check if the light scrollbar class is applied
    expect(scrollAreaElement).toHaveClass("scrollbar-light");
  });

  it("should apply the correct scrollbar class based on theme mode (dark)", () => {
    // Set mockTheme to dark before rendering
    mockTheme = { mode: "dark" };

    render(
      <ScrollArea maxHeight="96">
        <div>Dark Theme Child</div>
      </ScrollArea>
    );

    const scrollAreaElement = screen.getByText("Dark Theme Child").parentElement;

    // Check if the dark scrollbar class is applied
    expect(scrollAreaElement).toHaveClass("scrollbar-dark");
  });

  it("should render and match the snapshot", () => {
    const { asFragment } = render(
      <ScrollArea maxHeight="96">
        <div>Snapshot Test</div>
      </ScrollArea>
    );

    // Snapshot the ScrollArea structure
    expect(asFragment()).toMatchSnapshot();
  });
});
