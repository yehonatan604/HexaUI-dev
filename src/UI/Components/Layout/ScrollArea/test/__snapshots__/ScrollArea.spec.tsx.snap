// Vitest Snapshot v1

exports[`ScrollArea Component > should render and match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-96 overflow-auto scrollbar-light undefined"
  >
    <div>
      Snapshot Test
    </div>
  </div>
</DocumentFragment>
`;
